pkg/parser/parser.go

```diff
@@ -1373,6 +1393,8 @@ func (p *Parser) parseLit() ast.Expr {
 		return p.parseParam()
 	case "CASE":
 		return p.parseCaseExpr()
+	case "IF":
+		return p.parseIFExpr()
```

Comment:
Ditto here. `false OR IF(X < Y, true, false)` is valid, but `IF(X < Y, true, false) OR false` is invalid. Is it really?

I guess we should call `parseIfExpr` from `parseLit` instead.